### PR TITLE
Add udfs for creating and parsing custom build ids

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This are utility functions for https://github.com/mozilla/bigquery-etl/issues/1403

The `format_build_hour` udf is intended to be used in the `logical_app_id` views to convert the `app_build_id` into a custom id that throws away architecture information. The `parse_build_hour` udf is meant to be used in the `generate.py` file as the `build_date_udf`.

### CODE REVIEW

File: sql/mozfun/glam/parse_build_hour/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+description: |
+  Parses the custom build id used for Fenix builds in GLAM.
```
Comment (Ben-Wu) [inline sql/mozfun/glam/parse_build_hour/metadata.yaml:2]:
```suggestion
  Parses the custom build id used for Fenix builds in GLAM to a datetime.
```

File: sql/mozfun/glam/parse_build_hour/udf.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CREATE OR REPLACE FUNCTION glam.parse_build_hour(build_hour STRING)
```
Comment (Ben-Wu) [inline sql/mozfun/glam/parse_build_hour/udf.sql:1]: Something like `build_hour_to_datetime` would be more descriptive

File: sql/mozfun/glam/format_build_hour/udf.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
+CREATE OR REPLACE FUNCTION glam.format_build_hour(app_build_id STRING)
```
Comment (Ben-Wu) [inline sql/mozfun/glam/format_build_hour/udf.sql:1]:
```suggestion
CREATE OR REPLACE FUNCTION glam.fenix_build_to_build_hour(app_build_id STRING)
```

**Review by Ben-Wu — CHANGES_REQUESTED** [review]
I think the function names are a little confusing.

Also should we change `udf.fenix_build_to_datetime` to call the mozfun version?

**Comment by acmiyaguchi** [comment]
Thanks, these are certainly better names. We'll need to wait until the next deploy before we can reference the mozfun function in the udf module, but yes, we should.

File: sql/mozfun/glam/parse_build_hour/udf.sql (hunk 2 above)
Comment (Ben-Wu) [inline sql/mozfun/glam/parse_build_hour/udf.sql:1]:
```suggestion
CREATE OR REPLACE FUNCTION glam.build_hour_to_datetime(build_hour STRING)
```

**Review by Ben-Wu — APPROVED** [review]
Just the other renaming but otherwise good

**Pull request merged**